frontend-new/src/chat/chatMessage/typingChatMessage/TypingChatMessage.tsx

```diff
@@ -41,23 +43,27 @@ const textVariants = {
 };
 
 const TypingChatMessage: React.FC<TypingChatMessageProps> = ({ waitBeforeThinking = WAIT_BEFORE_THINKING }) => {
-  const [displayText, setDisplayText] = useState(UI_TEXT.TYPING);
+  const { t } = useTranslation();
+  const [isThinking, setIsThinking] = useState(false);
```

Comment:
Why use boolean over keys?

Keep the original implementation of constants.

```typescript
 const [displayTextKey, setDisplayTextKey] = useState(UI_TEXT.TYPING);
```

It is not that SI are thinking, it is that some specific time have passed (thinking took long).

see: `WAIT_BEFORE_THINKING`